_posts/2023-09-20-intro-to-the-various-java-based-k8s-oc-tools.adoc

```diff
@@ -0,0 +1,607 @@
+
+[source,xml]
+----
+<profile>
+    <id>jib</id>
+    <activation>
+        <property>
+            <name>image.builder</name>
+            <value>jib</value>
+        </property>
+    </activation>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.wildfly.plugins</groupId>
+                <artifactId>wildfly-maven-plugin</artifactId>
+                <configuration>
+                    <feature-packs>
+                        <feature-pack>
+                            <location>org.jboss.ws.cxf:jbossws-cxf-feature-pack:${project.version}</location>
+                        </feature-pack>
+                        <feature-pack>
+                            <location>org.wildfly:wildfly-galleon-pack:${jboss.version}</location>
+                        </feature-pack>
+                    </feature-packs>
+                    <layers>
+                        <layer>cloud-server</layer>
+                        <layer>webservices</layer>
+                    </layers>
+                    <filename>${warName}.war</filename>
+                </configuration>
+                <executions>
+                    <execution>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>package</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>com.google.cloud.tools</groupId>
+                <artifactId>jib-maven-plugin</artifactId>
+                <configuration>
+                    <from>
+                        <image>quay.io/wildfly/wildfly-runtime:latest</image>
+                    </from>
+                    <to>
+                        <image>${imageName}:${imageTag}</image>
+                    </to>
+                    <extraDirectories>
+                        <paths>
+                            <path>
+                                <from>target/server</from>
+                                <into>/opt/server</into>
+                            </path>
+                        </paths>
+                        <permissions>
+                            <permission>
+                                <file>/opt/server/**/*</file>
+                                <mode>770</mode>
+                            </permission>
+                        </permissions>
+                    </extraDirectories>
+                    <container>
+                        <user>root</user>
+                    </container>
+                </configuration>
+
+                <executions>
+                    <execution>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>dockerBuild</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+----
+
```

Comment:
Good writing for the jbossws part.  It looks good to me.

Reply:
Thanks!